reactioncommerce/reaction
# (wip)(refactor): Convert Products collection hooks to Hooks.Events

Resolves #3634

Use `Hooks.Events` for Products related hooks.

### Changes made
1.  `products.before.insert`: 
      - **Hooks.Events.add("verifyProductInsert"..)** 
2. `products.after.insert`: 
      - **Hooks.Events.add("afterProductInsert", (product)..)**
      - **Hooks.Events.add("afterProductInsertSearch", (product)..) **
3. `products.before.remove`:
     - **Hooks.Events.add("beforeProductRemove", (product)..)**
4. `products.after.remove`:
     - **Hooks.Events.add("afterProductVariantRemove", (product)..)**
     - **Hooks.Events.add("afterProductRemove", (product)..)**
5. `products.before.update`:
     - **Hooks.Events.add("beforeProductUpdate", (productUpdateArgs)..)**
6. `products.after.update`:
     - **Hooks.Events.add("afterProductUpdate", (productUpdateArgs)..)**
     - **Hooks.Events.add("afterProductUpdateSearchRebuild", (productUpdateArgs)..)**

Search: `imports/plugins/included/search-mongo/server/hooks/search.js`: No tests for this hook. You need to test this manually to make sure it works.

How to test
- Update product
- Make sure the search record has been rebuilt if the fields modified are part of the ones watched
